### ISSUE TITLE
Create Code Standard + Review Process

### ISSUE THREAD

**Issue by L00166967** [ruthlennonatu/bsc23Hurlfit#63]
As discussed in class, I'd like to be in charge of making sure the code we write is clean, quality and consistent so that we're better able to know how to look for each other's files, and to understand each other's code.

In order to do this effectively, we'd need to come up with a consistent set of code principles we can put in the readme, or the wiki, that our code can follow. That way, we can ensure our code looks and works similarly by following the guide, and we know it's clean enough if it follows said guide.

I'd like to be the code reviewer; code could for example be written in our branches, but also go through code review to see if it follows the guide before going to the main branch. There are also some ways to automate parts of this; I know Python has PEP for instance, and there are programs that can check code to see if it fits PEP.

I think a good few first rules to sort out would be:

- Finding the style guide for Javascript/React
- Naming conventions for variables/functions(do we go with `fetchFromDatabase` or `databaseFetcher`?)
- Never nester(not allowing us to nest within more than 3 pairs of braces before extracting to a helper function, helps keep complex code readable.)

But we'll need to discuss this as a team later. Ruth said this was good for the pipeline, and my passion lies in code. We also need to do this sooner rather than later because our own code will rely on this.

**Comment by L00166967** [comment]
Forgot to mention; I updated hte readme the same day I made this issue in the Documents branch.

**Comment by L00166967** [comment]
(Copied from slack)
I've looked at a variety of standards. They all have their strengths and weaknesses. I've narrowed it down to [google's](https://google.github.io/styleguide/jsguide.html#formatting-braces) , [idomatic js](https://github.com/rwaldron/idiomatic.js/) , or [standardjs](https://standardjs.com/) , and I think they all have linters so we can follow them as we write our code.

 - Regarding Google's coding standard, I appreciate its specific guidelines on function naming and the use of docstrings to clarify function parameters and return values.
 - Idiomatic JS stands out to me due to its simplicity and ease of understanding.
 - StandardJS, on the other hand, offers the added benefit of a precommit hook, ensuring that any committed code adheres to the standard. This would simplify the code review process for us; one less thing to worry about, and helps to ensure consistency.

In addition, should we not go with google's own style guide, we need to discuss function names and docstrings - I believe they will make it easier for us to read each other's code. I spend a lot of time trying to figure out what a function does at times.
